ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/RemoveUnusedNamedPatternFix.fs

```diff
@@ -0,0 +1,75 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open System
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Psi.ExtensionsAPI.Tree
+open JetBrains.ReSharper.Psi.Util
+open JetBrains.ReSharper.Resources.Shell
+open JetBrains.TextControl
+
+type RemoveUnusedNamedPatternFix(warning: UnusedValueWarning) =
+    inherit FSharpQuickFixBase()
+    
+    let fieldAndRecord =
+        match warning.Pat.IgnoreInnerParens().Parent with
+        | :? IFieldPat as fieldPat ->
+            match fieldPat.Parent with
+            | :? IRecordPat as recordPat ->
+                Some(fieldPat, recordPat)
+            | _ -> None
+        | _ -> None
+        
+    override x.Text = "Remove unused value"
+
+    override x.IsAvailable _ =
+        match fieldAndRecord with
+        | None -> false
+        | Some(fieldPat, recordPat) ->
+            isValid fieldPat && isValid recordPat
+
+    override x.ExecutePsiTransaction(_, _) =
+        match fieldAndRecord with
+        | None -> null
+        | Some(fieldPat, recordPat) ->
+
+        use writeLock = WriteLockCookie.Create(fieldPat.IsPhysical())
+        use disableFormatter = new DisableCodeFormatter()
+
+        let indexPat = recordPat.FieldPatterns.IndexOf(fieldPat)
+
+        if recordPat.FieldPatterns.Count = 1 then
+            let factory = recordPat.CreateElementFactory()
+            let wildPat = factory.CreateWildPat()
+            ModificationUtil.ReplaceChild(recordPat, wildPat) |> ignore
+            null
+            
+        elif indexPat < recordPat.FieldPatterns.Count - 1 then
+            let nextFieldPat = recordPat.FieldPatterns.[indexPat + 1]
+            
+            let rangeToDelete =
+                let rangeStart = fieldPat
+                let rangeEnd =
+                    nextFieldPat.PrevSibling
+                TreeRange(rangeStart, rangeEnd)
```

Comment:
This approach doesn't look at what could be between the field patterns, and there are some important cases to consider when using it:

We probably don't want to remove the new line between the fields here:
```fsharp
match r with
| { F1 = 1; F2 = f2
    F3 = 3; F4 = 4 } -> ()
```

While here we would not want to keep an empty line:
```fsharp
match r with
| { F1 = 1
    F2 = f2
    F3 = 3 } -> ()
```

What I'd suggest trying to do here is this:
* get the field pattern node
* check if it's the last field pattern on the line, i.e. if `fieldPat.EndLine <> nextPattern.StartLine`
  * if it's the last one: try to go left from it and possibly include the previous new line token, like in the last example, using `getFirstMatchingNodeBefore isInlineSpace fieldPat |> getThisOrPrevNewLine` helpers
* try to move to the right using `getLastMatchingNodeAfter isInlineSpace`

Please check existing usages of these helpers.

Reply:
Some more cases to add a test for:

```fsharp
match r with
| { F1 = 1; F2 = 2
    F3 = f3; F4 = 4 } -> ()
```

```fsharp
match r with
| { F1 = 1; F2 = 2
    F3 = f3 } -> ()
```

```fsharp
match r with
| { F1 = 1; F2 = f2 // comment
    F3 = 3 } -> ()
```